airdrop/infrastructure/repositories/claim_history_repo.py

```diff
@@ -17,3 +20,26 @@ def add_claim(self, claim_payload):
 
             )
         )
+
+    def get_pending_claims_for_given_airdrop_id(self, airdrop_id, blockchain_method):
+        claims = self.session.query(ClaimHistory). \
+            filter(ClaimHistory.airdrop_id == airdrop_id). \
+            filter(ClaimHistory.blockchain_method == blockchain_method). \
+            filter(ClaimHistory.transaction_status == AirdropClaimStatus.PENDING.value).all()
+        return claims
+
+    def update_claim_status(self, address, airdrop_window_id, blockchain_method, transaction_status,
+                            transaction_hash=None):
+        try:
+            claim = self.session.query(ClaimHistory) \
+                .filter(ClaimHistory.address == address) \
+                .filter(self.session.airdrop_window_id == airdrop_window_id) \
+                .filter(ClaimHistory.blockchain_method == blockchain_method) \
+                .first()
```

Comment:
we can combine all three filters in a single filter evaluation

Reply:
Final queries is same. Above add more viability instead separating it by comma's.